kpatch-build/create-diff-object.c

```diff
@@ -1933,6 +1970,30 @@ static void kpatch_migrate_included_elements(struct kpatch_elf *kelf, struct kpa
 			sec->secsym = NULL;
 	}
 
+#ifdef __aarch64__
+{
+	/*
+	 * On aarch64 the __patchable_function_entries section is used to store
+	 * information about patchable functions which we will use in
+	 * kpatch_create_mcount_sections(). Since this section isn't included it
+	 * will be freed in kpatch_elf_teardown(), so we preserve it in the main
+	 * function. However the relas->sym of any relocation will be freed if
+	 * that symbol isn't included and thus it is unsafe to dereference it
+	 * later. Zero the "sym" field for any symbol not included.
+	 */
+	struct rela *rela;
+	struct section *patchable_sec;
+
+	patchable_sec = find_section_by_name(&kelf->sections,
+					     "__patchable_function_entries");
+	if (patchable_sec) {
+		list_for_each_entry(rela, &patchable_sec->rela->relas, list) {
+			if (!rela->sym->include)
+				rela->sym = NULL;
+		}
+	}
+}
+#endif
```

Comment:
This feels like a major hack and I'm not really convinced it's needed.  I have a more detailed comment below.